app/decorators/user_decorator.rb

```diff
@@ -51,6 +51,13 @@ def active_or_pending_profile
     user.active_profile || pending_profile
   end
 
+  # This user's most recently activated profile that has also been deactivated
+  # due to a password reset, or nil if there is no such profile
+  def password_reset_profile
```

Comment:
wondering if this brings up any edge cases -- what if someone has requested a code via USPS for two profiles, for example?